File: Week_08/338/counting_bits.py
from typing import List


class Solution:
    def countBits(self, num: int) -> List[int]:
        res = [0] * (num + 1)
        for i in range(1, num + 1):
            # res[i] = (i & 1) + res[i >> 1]
            res[i] = i % 2 + res[i // 2]
        return res

File: Week_08/338/counting_bits.go
package main

func countBits(num int) []int {
	res := make([]int, num+1)
	for i:=1;i <= num; i++ {
		//res[i] = i % 2 + res[i /2]
		res[i] = i & 1 + res[i >> 1]
	}
	return res
}